core/src/main/java/com/expedia/www/haystack/client/Tracer.java

```diff
@@ -14,14 +17,35 @@
 import io.opentracing.util.ThreadLocalActiveSpanSource;
 
 public class Tracer implements io.opentracing.Tracer {
+    private final Dispatcher dispatcher;
     private final Clock clock;
     private final String serviceName;
     private final ActiveSpanSource activeSource;
 
-    public Tracer(String serviceName, ActiveSpanSource activeSource, Clock clock) {
+    public Tracer(String serviceName, ActiveSpanSource activeSource, Clock clock, Dispatcher dispatcher) {
         this.serviceName = serviceName;
         this.activeSource = activeSource;
         this.clock = clock;
+        this.dispatcher = dispatcher;
+    }
+
+    public void close() throws IOException {
+        dispatcher.close();
+    }
+
+    public void flush() throws IOException {
+        dispatcher.flush();
+    }
+
+    void dispatch(Span span) {
+        dispatcher.dispatch(span);
+    }
+
+    /**
+     * @return the dispatcher
+     */
+    public Dispatcher getDispatcher() {
```

Comment:
Do you really need to expose the Dispatcher via getDispatcher? I hate getters unless they're in classes used for transport.

Reply:
I added this for a specific reason, it escapes me at this moment, but I knew it was important to have access to.  This might change to package protection in a latter PR.